cluster-autoscaler/core/scale_down.go

```diff
@@ -631,24 +618,18 @@ func (sd *ScaleDown) isNodeBelowUtilzationThreshold(node *apiv1.Node, utilInfo s
 // updateUnremovableNodes updates unremovableNodes map according to current
 // state of the cluster. Removes from the map nodes that are no longer in the
 // nodes list.
-func (sd *ScaleDown) updateUnremovableNodes(nodes []*schedulernodeinfo.NodeInfo) {
+func (sd *ScaleDown) updateUnremovableNodes() {
 	if len(sd.unremovableNodes) <= 0 {
 		return
 	}
-	// A set of nodes to delete from unremovableNodes map.
-	nodesToDelete := make(map[string]struct{}, len(sd.unremovableNodes))
-	for name := range sd.unremovableNodes {
-		nodesToDelete[name] = struct{}{}
-	}
-	// Nodes that are in the cluster should not be deleted.
-	for _, node := range nodes {
-		if _, ok := nodesToDelete[node.Node().Name]; ok {
-			delete(nodesToDelete, node.Node().Name)
+	newUnremovableNodes := make(map[string]time.Time, len(sd.unremovableNodes))
+	for oldUnremovable, since := range sd.unremovableNodes {
+		if _, err := sd.context.ClusterSnapshot.NodeInfos().Get(oldUnremovable); err == nil {
```

Comment:
Log on error here

Reply:
Added logging on info level - this is supposed to happen once a node is removed.